packages/clicker-game/lib/game.js

```diff
@@ -1,42 +1,50 @@
-const defaults = require('lodash/defaults')
+const defaults = require("lodash/defaults")
 
 class Game {
   constructor(state = {}) {
     this.state = defaults({}, state, {
       initialScore: 0,
+      passiveIncome: 0,
       actions: [],
       lastSynced: null
     })
     this.state.lastSynced = new Date(this.state.lastSynced)
+    this.click = this.click.bind(this);
+    this.validate = this.validate.bind(this);
   }
 
   pushAction(action) {
     const timestamp = new Date()
     this.state.actions.push({ action, timestamp })
   }
 
-  click() {
-    this.pushAction("click")
+  click(action) {
+    if(!this.validate(action)) return false;
```

Comment:
Clicking should just push the action to the action queue. The resulting state will then be reflected in `getCurrentState`